exporter/kafkaexporter/internal/kafkaclient/franzgo.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kafkaclient // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter/internal/kafkaclient"
+
+import (
+	"context"
+	"errors"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+// FranzSyncProducer is a wrapper around the franz-go client that implements
+// the Producer interface. Allowing us to use the franz-go client while
+// maintaining compatibility with the existing Kafka exporter code.
+type FranzSyncProducer struct {
+	client       *kgo.Client
+	metadataKeys []string
+}
+
+// NewFranzSyncProducer Franz-go producer from a kgo.Client and a Messenger.
+func NewFranzSyncProducer(client *kgo.Client,
+	metadataKeys []string,
+) FranzSyncProducer {
+	return FranzSyncProducer{
+		client:       client,
+		metadataKeys: metadataKeys,
+	}
+}
+
+// ExportData sends a batch of messages to Kafka
+func (p FranzSyncProducer) ExportData(ctx context.Context, msgs Messages) error {
```

Comment:
Should we make these pointer receivers, to avoid copying fields on each invocation?